### PR TITLE
Add Github Actions CI test on Windows

### PR DESCRIPTION
**Pull request by madkinsz**
## Summary

We want to move away from AppVeyor and this sets up a comprehensive test using GithubActions

Will require AppVeyor to be removed from the repo after merge

## Changes

- Replaces AppVeyor with testing on Windows via Github Actions
- Tests additional python versions on Windows
- Some tests using `parse_path` were failing and I do not see how they would ever succeed on Windows; skips have been added to the relevant cases.
- Some tests are brittle on the Github Actions Windows runner and have been slightly modified

## Future work

- We could simplify our CI by rolling the Circle CI Unix tests into this workflow with a simple matrix

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -195,7 +195,7 @@ jobs:
           command: pytest tests/tasks -vvrfEsx --numprocesses 4 --dist=loadfile
 
   # ----------------------------------
-  # Run unit tests in Python 3.6-3.8
+  # Run unit tests in Python 3.6-3.9
```
Comment (mashun4ek) [inline .circleci/config.yml:198]:
```suggestion
  # Run unit tests in Python 3.7-3.9
```

Comment (madkinsz) [reply .circleci/config.yml:198]: We still test 3.6 on Unix; I just did not include it in the Windows tests

Comment (mashun4ek) [reply .circleci/config.yml:198]: I see, makes sense

File: .github/workflows/test_windows.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: Run tests on Windows
+
+on: [push]
```
Comment (zangell44) [inline .github/workflows/test_windows.yml:3]: do we want to just tigger this for pull requests?

File: tests/executors/test_executors.py [hunk 3]
```diff
@@ -178,37 +178,32 @@ def test_interrupt_stops_running_tasks_quickly(self, scheduler):
...
-            pass
-        except Exception:
-            assert False, "Failed to interrupt"
+            pass  # Don't exit test on the interrupt
+
         stop = time.time()
-        if stop - start > 4:
-            assert False, "Failed to interrupt"
+
+        # Defining "quicky" here as 4 seconds generally and 6 seconds on
```
Comment (zangell44) [inline tests/executors/test_executors.py:204]: `"quickly"`

File: tests/run_configs/test_ecs.py [hunk 4]
```diff
@@ -82,13 +81,13 @@ def test_local_task_definition_path(tmpdir, scheme):
         ]
     }
     path = str(tmpdir.join("test.yaml"))
+    print(repr(path))
```
Comment (zangell44) [inline tests/run_configs/test_ecs.py:84]: can we remove?

**Review by zangell44 — COMMENTED** [review]
Couple comments/questions

File: tests/run_configs/test_ecs.py (hunk 4 above)
Comment (madkinsz) [reply tests/run_configs/test_ecs.py:84]: Yep
```suggestion
```

File: tests/executors/test_executors.py (hunk 3 above)
Comment (madkinsz) [reply tests/executors/test_executors.py:204]:
```suggestion
        # Defining "quickly" here as 4 seconds generally and 6 seconds on
```

File: .github/workflows/test_windows.yml (hunk 2 above)
Comment (madkinsz) [reply .github/workflows/test_windows.yml:3]: Ah I don't know how off the top of my head. `pull_request` is more than that https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request

Comment (madkinsz) [reply .github/workflows/test_windows.yml:3]: Nevermind the default types cover pushes to PRs

Comment (madkinsz) [reply .github/workflows/test_windows.yml:3]:
```suggestion
on: [pull_request]
```

**Review by zangell44 — APPROVED** [review]

**Pull request merged**